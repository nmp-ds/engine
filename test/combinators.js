import { setup } from './_helpers.js';
import { describe, expect, test } from 'vitest';

setup();

describe('combinators', () => {
  test('all', async ({ uno }) => {
    const classes = ['all:mt-32', 'all:-mt-32', 'all:-bottom-1/2', '!all:mt-32'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('children', async ({ uno }) => {
    const classes = ['children:mt-32', 'children:-mt-32', 'children:-bottom-1/2', '!children:mt-32'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('next', async ({ uno }) => {
    const classes = ['next:mt-32', 'next:-mt-32', 'next:-bottom-1/2', '!next:mt-32'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('sibling', async ({ uno }) => {
    const classes = ['sibling:mt-32', 'sibling:-mt-32', 'sibling:-bottom-1/2', '!sibling:mt-32'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('siblings', async ({ uno }) => {
    const classes = ['siblings:mt-32', 'siblings:-mt-32', 'siblings:-bottom-1/2', '!siblings:mt-32'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });

  // space/divide variant is already tested by the space/divide classes
});
