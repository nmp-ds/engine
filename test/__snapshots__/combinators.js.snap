// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`combinators > all 1`] = `
"/* layer: default */
.all\\:-bottom-1\\/2 *{bottom:-50%;}
.\\!all\\:mt-32 *{margin-top:3.2rem !important;}
.all\\:-mt-32 *{margin-top:-3.2rem;}
.all\\:mt-32 *{margin-top:3.2rem;}"
`;

exports[`combinators > children 1`] = `
"/* layer: default */
.children\\:-bottom-1\\/2>*{bottom:-50%;}
.\\!children\\:mt-32>*{margin-top:3.2rem !important;}
.children\\:-mt-32>*{margin-top:-3.2rem;}
.children\\:mt-32>*{margin-top:3.2rem;}"
`;

exports[`combinators > next 1`] = `
"/* layer: default */
.next\\:-bottom-1\\/2+*{bottom:-50%;}
.\\!next\\:mt-32+*{margin-top:3.2rem !important;}
.next\\:-mt-32+*{margin-top:-3.2rem;}
.next\\:mt-32+*{margin-top:3.2rem;}"
`;

exports[`combinators > sibling 1`] = `
"/* layer: default */
.sibling\\:-bottom-1\\/2+*{bottom:-50%;}
.\\!sibling\\:mt-32+*{margin-top:3.2rem !important;}
.sibling\\:-mt-32+*{margin-top:-3.2rem;}
.sibling\\:mt-32+*{margin-top:3.2rem;}"
`;

exports[`combinators > siblings 1`] = `
"/* layer: default */
.siblings\\:-bottom-1\\/2~*{bottom:-50%;}
.\\!siblings\\:mt-32~*{margin-top:3.2rem !important;}
.siblings\\:-mt-32~*{margin-top:-3.2rem;}
.siblings\\:mt-32~*{margin-top:3.2rem;}"
`;
